### DISCUSSION TITLE
ControlTemplate ignored on ContentPage

### DISCUSSION THREAD

**Discussion by mpanzica** [dotnet/maui#5687] [Q&A]
Hello,

I'm trying to use ControlTemplate in a ContentPage.  What I used to do on Xamarin does not seem to be working on MAUI.
Using the code below, I always get a full yellow box (i.e.: no red and fuchsia boxes)
Is this a bug or am I doing wrong here?

(I'm attaching a screenshot since pasted code does not get displayed properly. Sorry!)

<img width="483" alt="snippet" src="https://user-images.githubusercontent.com/26141121/162577974-c845928b-2107-48dc-b1de-156d6098d5a6.png">

**Comment by BinaryAssault** [comment]
For fun, have you tried defining the control template in app.xaml and removing from your page? I've never seen a control template in a content page, defined for the content page (doesn't mean you're wrong). I wonder if the order of initialization is what is causing your problem.

**Reply by ghost** [reply]
Yes, I've tried of course. Still the same behaviour.
The reason why I define the ControlTemplate in the ContentPage instead of the App.xaml is to make a specific type of page self containing and reusable across different projects.
Please not that the exact same xaml (1st namespace fixed) works like a charm on Xamarin Forms.

Just found out that issue #2302 reported the exact same bug but has been closed without addressing the issue.

**Reply by hartez** [reply]
#2302 shouldn't have been closed - I've re-opened it.

**Reply by ghost** [reply]
Thanks hartez!

**Reply by pictos** [reply]
@hartez we can't comment on the issue, just members can comment. Is that intentional?

**Reply by hartez** [reply]
@pictos I think it's because the robot locks comments on issues after they've been resolved for a while (to prevent spam) but doesn't re-enable them when the issue is re-opened.

Anyway, #6428 should fix this issue.
